src/releaseNotesProvider.ts

```diff
@@ -0,0 +1,180 @@
+import * as vscode from "vscode";
+import * as semver from "semver";
+import { Remarkable } from "remarkable";
+import http from "https";
+
+const versionKey = "metals-server-version";
+
+// prettier-ignore
+const releaseNotesMarkdownUrl: Record<string, string> = {
+  "0.11.5": "https://raw.githubusercontent.com/scalameta/metals/main/website/blog/2022-04-28-aluminium.md",
+  "0.11.6": "https://raw.githubusercontent.com/scalameta/metals/main/website/blog/2022-06-03-aluminium.md",
+};
+
+/**
+ * Show release notes if possible, swallow errors since its not a crucial feature.
+ * Treats snapshot versions like 0.11.6+67-926ec9a3-SNAPSHOT as a 0.11.6.
+ */
+export async function showReleaseNotesIfNeeded(
+  context: ExtensionContext,
+  serverVersion: string,
+  outputChannel: vscode.OutputChannel
+) {
+  try {
+    await showReleaseNotes(context, serverVersion);
+  } catch (error) {
+    outputChannel.appendLine(
+      `Couldn't show release notes for Metals ${serverVersion}`
+    );
+    outputChannel.appendLine(`${error}`);
+  }
+}
+
+async function showReleaseNotes(
+  context: ExtensionContext,
+  currentVersion: string
+): Promise<void> {
+  const state = context.globalState;
+
+  const version = getVersion();
+  const releaseNotesUrl = version
+    ? releaseNotesMarkdownUrl[version]
+    : undefined;
+  if (isNotRemote() && version && releaseNotesUrl) {
+    await showPanel(version, releaseNotesUrl);
+  }
+
+  async function showPanel(version: string, releaseNotesUrl: string) {
+    const releaseNotes = await getReleaseNotesMarkdown(releaseNotesUrl);
+
+    if (typeof releaseNotes === "string") {
+      const panel = vscode.window.createWebviewPanel(
+        `scalameta.metals.whatsNew`,
+        `Metals ${version} release notes`,
+        vscode.ViewColumn.One
+      );
+
+      panel.webview.html = releaseNotes;
+      panel.reveal();
+
+      // set wrong value for local development
+      // const newVersion = "0.11.5";
+
+      const newVersion = version;
+
+      // Update current device's latest server version when there's no value or it was a older one.
+      // Then sync this value across other devices.
+      state.update(versionKey, newVersion);
+      state.setKeysForSync([versionKey]);
+
+      context.subscriptions.push(panel);
+    }
+  }
+
+  /**
+   * Don't show panel for remote environment because it installs extension on every time.
+   * TODO: what about wsl?
+   */
+  function isNotRemote(): boolean {
+    const isNotRemote = env.remoteName == null;
+    // const isWsl = env.remoteName === "wsl";
+    return isNotRemote;
+  }
+
+  /**
+   *  Return version for which release notes should be displayed
+   *  or
+   *  undefined if notes shouldn't be displayed
+   */
+  function getVersion(): string | undefined {
+    const previousVersion: string | undefined = state.get(versionKey);
+    // strip version to 'major.minor.patch'
+    const cleanVersion = semver.clean(currentVersion);
+    if (cleanVersion) {
```

Comment:
`semver.clean(currentVersion)` will return 'major.minor.patch'. Technically it can return null/undefined if input is invalid but `currentVersion` should be valid Metals version, otherwise downloading it will fail in `extension.ts`